source/ring_buffer.c

```diff
@@ -43,24 +47,33 @@ void aws_ring_buffer_clean_up(struct aws_ring_buffer *ring_buf) {
 }
 
 int aws_ring_buffer_acquire(struct aws_ring_buffer *ring_buf, size_t requested_size, struct aws_byte_buf *dest) {
+    AWS_PRECONDITION(aws_ring_buffer_is_valid(ring_buf));
+    AWS_PRECONDITION(aws_byte_buf_is_valid(dest));
     if (requested_size == 0) {
+        AWS_POSTCONDITION(aws_ring_buffer_is_valid(ring_buf));
```

Comment:
These postconditions seem unnecessary since `ring_buf` and `dest` have not been touched since the precondition.

My intuition is to keep them in case we ever start generating documentation from these postconditions. So I'm not going to request that you change this, especially since you've noted that these checks are cheap.